src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.ReferenceProcessing.cs

```diff
@@ -121,6 +128,18 @@ static bool IsConstructorForType(ISymbol possibleConstructor, ISymbol possibleTy
             }
         }
 
+        private static bool IsPropertyGeneratedFromPrimaryConstructorParameter(
+            ISymbol propertySymbol, ISymbol parameterSymbol, CancellationToken cancellationToken)
+        {
+            if (parameterSymbol is IParameterSymbol parameter)
+            {
+                var generatedPropertySymbol = parameter.GetAssociatedSynthesizedRecordProperty(cancellationToken);
+                return propertySymbol.Equals(generatedPropertySymbol);
+            }
+
+            return false;
```

Comment:
nit: can be written as:

`=> parameterSymbol is IParameterSymbol parameter && propertySymbol.Equals(parameter.GetAssociatedSynthesizedRecordProperty(cancellationToken));`